core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -92,7 +92,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
     self.sandbox
       .foldM(
         cause => ZIO.fail(cause.squashWith(f)),
-        ZIO.succeed
+        ZIO.succeed(_)
```

Comment:
Were these breaking compilation? If not, please revert all those changes to minimise the diff.

Reply:
I think unfortunately these are necessary because a by name function is different than a by value one to the compiler, so we need to explicitly apply the function.